File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
function square_of_sum(n)
    #sum(1:n)^2
    ((n^2 + n) ÷ 2)^2
end

"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
    #sum(x -> x^2, 0:n)
    (n * (n + 1) * (2n + 1)) ÷ 6
end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
function difference(n)
    square_of_sum(n) - sum_of_squares(n)
end

File: solutions/go/difference-of-squares/1/difference_of_squares.go
/*
Package diffsquares implements functions to pass the Exercism test cases
*/
package diffsquares

import "math"

// SquareOfSums returns the square of the sum of numbers up to number
func SquareOfSums(number int) int {
	sum := 0
	for i := 1; i <= number; i++ {
		sum += i
	}
	return int(math.Pow(float64(sum), 2))
}

// SumOfSquares returns the sum squared of the numbers up to number
func SumOfSquares(number int) int {
	sum := 0.0
	for i := 1; i <= number; i++ {
		sum += math.Pow(float64(i), 2)
	}
	return int(sum)
}

// Difference returns the difference of SquareOfSum - SumOfSquares
func Difference(number int) int {
	a := SquareOfSums(number)
	b := SumOfSquares(number)
	return a - b
}